mono/metadata/mono-perfcounters.c

```diff
@@ -1482,17 +1490,77 @@ typedef struct {
 	int type;
 } CounterCreationData;
 
+TYPED_HANDLE_DECL (CounterCreationData);
+
+static char*
+append (char* cursor, const char* str)
+{
+	int const length = strlen (str) + 1;
+	memcpy (cursor, str, length);
+	return cursor + length;
+}
+
+static void
+System_Diagnostics_PerformanceCounterCategory_Create_create_temp (
+	MonoArrayHandle items, int i, char **counter_info, MonoError *error)
+// Avoid creating co-op handles in loops -- move loop body into functions.
+{
+	HANDLE_FUNCTION_ENTER ()
+
+	// FIXME Move this out of loop?
+	MONO_HANDLE_LOCAL_VARIABLE_INITIALIZED_NULL (CounterCreationData, data);
+	MONO_HANDLE_ARRAY_GETREF (data, items, i);
```

Comment:
Maybe we need a new macro so we could do this in one step: `CounterCreationDataHandle data = MONO_HANDLE_ARRAY_NEW_GETREF (CounterCreationData, items, i);` ?
(there's already something like this for fields, c.f. `MONO_HANDLE_NEW_GET(fieldType, handle, fieldName)`)